openff/interchange/drivers/report.py

```diff
@@ -52,7 +52,8 @@ def energies_validator(value: dict[str, Quantity | None]) -> dict[str, Quantity
         else:
             raise InvalidEnergyError(f"Energy type {key} not understood.")
 
-    return value
+    # return sorted in a consistent order
+    return {key: value[key] for key in _KNOWN_ENERGY_TERMS if key in value}
```

Comment:
Again, `pop`ping here and raising an error if there are unknown terms might be useful defensive coding.